File: iosApp/iosApp/Presentation/Main/Admin/Dormant/DormantAccountUiState.swift
import Foundation
import Shared

struct DormantAccountUiState {
    var selectedFilter: DormantAccountFilter = .dormant
    var users: [User] = []
    var nextCursor: String? = nil
    var canLoadMore: Bool = false
    var isLoading: Bool = false
    var isLoadingMore: Bool = false
    var isUpdating: Bool = false
    var confirmTarget: DormantChangeRequest? = nil
    var infoMessage: String? = nil

    var filterChips: [DormantAccountFilterChip] {
        [DormantAccountFilter.dormant, DormantAccountFilter.candidate].map { filter in
            DormantAccountFilterChip(
                filter: filter,
                label: filter.label,
                isSelected: selectedFilter == filter
            )
        }
    }
}

struct DormantAccountFilterChip: Identifiable {
    var id: DormantAccountFilter { filter }
    let filter: DormantAccountFilter
    let label: String
    let isSelected: Bool
}

struct DormantChangeRequest {
    let user: User
    let dormant: Bool
}

extension DormantAccountFilter {
    var label: String {
        switch self {
        case .dormant: return "휴면 계정"
        case .candidate: return "휴면 예정"
        default: return "계정"
        }
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/admin/dormant/DormantAccountUiState.kt
package com.hhp227.concafe.presentation.main.admin.dormant

import com.hhp227.concafe.domain.model.DormantAccountFilter
import com.hhp227.concafe.domain.model.User

data class DormantAccountUiState(
    val selectedFilter: DormantAccountFilter = DormantAccountFilter.DORMANT,
    val users: List<User> = emptyList(),
    val nextCursor: String? = null,
    val canLoadMore: Boolean = false,
    val isLoading: Boolean = false,
    val isLoadingMore: Boolean = false,
    val isUpdating: Boolean = false,
    val confirmTarget: DormantChangeRequest? = null,
    val infoMessage: String? = null
) {
    val filterChips: List<DormantAccountFilterChip>
        get() = DormantAccountFilter.entries.map { filter ->
            DormantAccountFilterChip(
                filter = filter,
                label = filter.label,
                isSelected = selectedFilter == filter
            )
        }
}

data class DormantAccountFilterChip(
    val filter: DormantAccountFilter,
    val label: String,
    val isSelected: Boolean
)

data class DormantChangeRequest(
    val user: User,
    val dormant: Boolean
)

val DormantAccountFilter.label: String
    get() = when (this) {
        DormantAccountFilter.DORMANT -> "휴면 계정"
        DormantAccountFilter.CANDIDATE -> "휴면 예정"
    }
